pkg/sync/originatorStream.go

```diff
@@ -2,75 +2,54 @@ package sync
 
 import (
 	"context"
-	"database/sql"
 	"errors"
 	"io"
 	"strings"
-	"time"
 
 	"github.com/cenkalti/backoff/v5"
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/xmtp/xmtpd/pkg/currency"
-	"github.com/xmtp/xmtpd/pkg/db"
-	"github.com/xmtp/xmtpd/pkg/db/queries"
 	envUtils "github.com/xmtp/xmtpd/pkg/envelopes"
-	"github.com/xmtp/xmtpd/pkg/fees"
 	"github.com/xmtp/xmtpd/pkg/metrics"
-	"github.com/xmtp/xmtpd/pkg/payerreport"
 	"github.com/xmtp/xmtpd/pkg/proto/xmtpv4/envelopes"
 	"github.com/xmtp/xmtpd/pkg/proto/xmtpv4/message_api"
 	"github.com/xmtp/xmtpd/pkg/registry"
-	"github.com/xmtp/xmtpd/pkg/topic"
-	"github.com/xmtp/xmtpd/pkg/utils"
 	"go.uber.org/zap"
 )
 
 type originatorStream struct {
-	ctx                        context.Context
-	db                         *sql.DB
-	log                        *zap.Logger
-	node                       *registry.Node
-	queries                    *queries.Queries
-	lastEnvelope               *envUtils.OriginatorEnvelope
-	stream                     message_api.ReplicationApi_SubscribeEnvelopesClient
-	feeCalculator              fees.IFeeCalculator
-	payerReportStore           payerreport.IPayerReportStore
-	payerReportDomainSeparator common.Hash
+	ctx          context.Context
+	log          *zap.Logger
+	node         *registry.Node
+	lastEnvelope *envUtils.OriginatorEnvelope
+	stream       message_api.ReplicationApi_SubscribeEnvelopesClient
+	writeQueue   chan *envUtils.OriginatorEnvelope
 }
 
 func newOriginatorStream(
 	ctx context.Context,
-	db *sql.DB,
 	log *zap.Logger,
 	node *registry.Node,
 	lastEnvelope *envUtils.OriginatorEnvelope,
 	stream message_api.ReplicationApi_SubscribeEnvelopesClient,
-	feeCalculator fees.IFeeCalculator,
-	payerReportStore payerreport.IPayerReportStore,
-	payerReportDomainSeparator common.Hash,
+	writeQueue chan *envUtils.OriginatorEnvelope,
 ) *originatorStream {
 	return &originatorStream{
 		ctx: ctx,
-		db:  db,
 		log: log.With(
 			zap.Uint32("originator_id", node.NodeID),
 			zap.String("http_address", node.HttpAddress),
 		),
-		node:                       node,
-		queries:                    queries.New(db),
-		lastEnvelope:               lastEnvelope,
-		stream:                     stream,
-		feeCalculator:              feeCalculator,
-		payerReportStore:           payerReportStore,
-		payerReportDomainSeparator: payerReportDomainSeparator,
+		node:         node,
+		lastEnvelope: lastEnvelope,
+		stream:       stream,
+		writeQueue:   writeQueue,
 	}
 }
 
 func (s *originatorStream) listen() error {
 	recvChan := make(chan *message_api.SubscribeEnvelopesResponse)
-	errChan := make(chan error)
-	writeQueue := make(chan *envUtils.OriginatorEnvelope, 10)
+	errChan := make(chan error, 1)
```

Comment:
guarantees that the reader routine can exit if the main routine is doing something else. Otherwise the write to err might block forever.